enterprise/frontend/src/embedding-sdk/components/public/CollectionBrowser/CollectionBrowser.tsx

```diff
@@ -0,0 +1,88 @@
+import { useState } from "react";
+
+import { withPublicComponentWrapper } from "embedding-sdk/components/private/PublicComponentWrapper";
+import { COLLECTION_PAGE_SIZE } from "metabase/collections/components/CollectionContent";
+import { CollectionItemsTable } from "metabase/collections/components/CollectionContent/CollectionItemsTable";
+import { isNotNull } from "metabase/lib/types";
+import CollectionBreadcrumbs from "metabase/nav/containers/CollectionBreadcrumbs/CollectionBreadcrumbs";
+import type {
+  CollectionEssentials,
+  CollectionId,
+  CollectionItem,
+  CollectionItemModel,
+} from "metabase-types/api";
+
+const USER_FACING_ENTITY_NAMES = [
+  "collection",
+  "dashboard",
+  "question",
+  "model",
+] as const;
+
+type UserFacingEntityName = typeof USER_FACING_ENTITY_NAMES[number];
+
+const ENTITY_NAME_MAP: Partial<
+  Record<UserFacingEntityName, CollectionItemModel>
+> = {
+  collection: "collection",
+  dashboard: "dashboard",
+  question: "card",
+  model: "dataset",
+};
+
+type CollectionBrowserProps = {
+  collectionId?: CollectionId;
+  onClick?: (item: CollectionItem) => void;
+  pageSize?: number;
+  visibleEntityTypes?: UserFacingEntityName[];
+};
+
+export const CollectionBrowserInner = ({
+  collectionId,
+  onClick,
+  pageSize = COLLECTION_PAGE_SIZE,
+  visibleEntityTypes = [...USER_FACING_ENTITY_NAMES],
```

Comment:
Yeah, it was more to satisfy typescript since it doesn't allow me to use a `readonly` array here. If I remove `as const` from `USER_FACING_ENTITY_NAMES`, it loses the type of `"question" | "collection" | ...` and just reverts to `string[]`

Reply:
oh okay. That's new, I don't come across this often. Seems fine to leave it as is then.

Reply:
What do you think? I'd like to preserve type safety here but it is a very annoying thing

Reply:
Ah sorry github being github. I'll leave it as is